presto-parser/src/main/java/com/facebook/presto/sql/parser/AstBuilder.java

```diff
@@ -1542,6 +1542,21 @@ private static String getType(SqlBaseParser.TypeContext type)
             return "MAP(" + getType(type.type(0)) + "," + getType(type.type(1)) + ")";
         }
 
+        SqlBaseParser.RowTypeContext rowType = type.rowType();
+        if (rowType != null && rowType.identifier().size() == rowType.type().size()) {
```

Comment:
Why do you check the sizes? They should always be the same (or else, there's a bug in the grammar). If anything, it should be an assertion.